Sources/NIO/LengthFieldBasedFrameDecoder.swift

```diff
@@ -0,0 +1,85 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+/// A `ChannelInboundHandler` used to decode length-delimited frames.
+///
+/// This is a protocol that has messages prefixed with an encoding of their length.
+/// Such a protocol would look like this:
+/// ````
```

Comment:
- isn't this one backtick too many?
- super minor: could we have an empty line (just `///`) between the code blocks?